lib/valueflow.cpp

```diff
@@ -4971,7 +4971,7 @@ static void valueFlowLifetime(TokenList &tokenlist, ErrorLogger *errorLogger, co
                 valueFlowForwardLifetime(tok, tokenlist, errorLogger, settings);
         }
         // address of
-        else if (tok->isUnaryOp("&")) {
+        else if (tok->isUnaryOp("&") && !Token::simpleMatch(tok->astParent(), "*")) {
```

Comment:
Should this check be inside the if-scope so it doesn't proceed checking the following conditions?

Reply:
Yes, done.